packages/wire/lib/messages/ClosingSignedMessage.ts

```diff
@@ -0,0 +1,61 @@
+import { BufferReader, BufferWriter } from "@node-lightning/bufio";
+import { ChannelId } from "@node-lightning/core";
+import { MessageType } from "../MessageType";
+import { IWireMessage } from "./IWireMessage";
+
+
+/**
+ */
```

Comment:
Add comment on the purpose of this message and include when we send it and when we expect to receive it.

Reply:
Looks good, but let's add two more things:
- purpose of fee negotiation
- how fee rate is convergent

Reply:
1. Maybe I am wrong isn't the purpose for fee negotiation is that how fast one of the two side wants the transaction
    to be included in the block?
2. As for the Fee rate convergence, On every negotiation step we must give up some amount from our proposal towards the other peer’s proposal. Eg. assuming the peer proposes a closing fee of 3000 satoshi and our estimate shows it must be 4000. “10”: our next proposal will be 4000-10=3990. This exchange continues using the channelID until both agree on the same fee or when one side fails the channel.

Should I update the nxt commit using the above point?

Reply:
1. Maybe I am wrong, isn't the purpose for fee negotiation is that how fast one of the two side wants the transaction to be included in the block?
2. As for the Fee rate convergence, On every negotiation step we must give up some amount from our proposal towards the other peer’s proposal. Eg. assuming the peer proposes a closing fee of 3000 satoshi and our estimate shows it must be 4000. “10”: our next proposal will be 4000-10=3990. This exchange continues using the channelID until both agree on the same fee or when one side fails the channel.

Should I update the nxt commit using the above points?

References :
1. https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#closing-negotiation-closing_signed
2. https://lightning.readthedocs.io/lightning-close.7.html

Reply:
Re 1: For mutual close both sides want the transaction in with minimal fees. Without anchors, the feerate_per_kw set at open or through update_fee is suggested to be 5x the amount that will get the transaction included in the block, which is extremely expensive. One of the benefits of mutual close is that the fee rate can be negotiated down from the existing fee rate to a more reasonable one.  

Re 2: Correct, though you may want to say that it is convergent by requiring the fee rate to be strictly between the last proposed and the counterparty's last proposed value. The back and forth continues until values are equal, at which point both sides can broadcast the closing tx.

Reply:
That's very informative. Wasn't aware that initial fee kept or suggested is that high! Thanks